presto-main/src/main/java/com/facebook/presto/sql/planner/plan/IntersectNode.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.plan;
+
+import com.facebook.presto.sql.planner.Symbol;
+import com.facebook.presto.sql.tree.SetOperation;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ListMultimap;
+
+import javax.annotation.concurrent.Immutable;
+
+import java.util.List;
+
+import static com.facebook.presto.sql.tree.SetOperation.SetOperationType.INTERSECT;
+
+@Immutable
+public class IntersectNode
+        extends SetOperationNode
+{
+    public IntersectNode(
+            @JsonProperty("id") PlanNodeId id,
+            @JsonProperty("sources") List<PlanNode> sources,
+            @JsonProperty("symbolMapping") ListMultimap<Symbol, Symbol> symbolMapping,
+            @JsonProperty("outputs") List<Symbol> outputs)
+    {
+        super(id, sources, symbolMapping, outputs);
+    }
+
+    @Override
```

Comment:
Instead of requiring subclasses to implement this method, add the type as an argument to SetOperationNode's constructor, pass the value in the call to super(), and implement this method in the parent class.